spec.mediawiki

```diff
@@ -1146,59 +1322,105 @@ transaction. They broadcast their transactions.
-their own, the taker will be given the option to place a new market order.
+In the event that the taker fails to respond to the maker's initialization 
+transaction, the maker will incur no violation.
+
+====Rule 2: An account's cancellation ratio must remain below the threshold
+
+The cancellation ratio is the ratio of the count of cancelled orders to the 
+count of completed orders on an account. The cancellation threshold is set
+by the DEX operator.
+An order is considered completed when all matches have fully settled. 
+An order is considered cancelled when a cancel order is matched to a standing 
+limit order, or when a connection is dropped before an order is filled.
+Cancellation of a partially filled order is counted as a full cancellation. 
+The cancellation ratio is evaluated on a 25-order rolling window.
 
-'''Rule 2: Your cancellation ratio must at or below 1/5'''
+There may be cases when the order is revoked by the DEX, e.g. during an 
+outage or while performing maintenance. In this case, no cancellations or 
+completions are added to the account statistics.
 
-This is the ratio of the count of cancelled orders to the count of completed 
-orders. An order is considered completed for a client once an initialization 
-transaction is broadcast. In the case that the client is the maker and the taker
-(first to act) fails to broadcast their initialization, the order is not 
-considered in calculating the ratio. 
+====Rule 3: Transaction outputs must be properly sized
 
-This rule is not in effect until a user has 5 orders to consider. This ensures 
-that a client can potentially cancel their first four orders without penalty, 
-as long as their fifth order is not cancelled. 
+The swap output value must be sized to exactly the matched amount. The fee rate
+must be at least the minimum value set by the DEX. It is the client's 
+responsibility to ensure that fees on a partial fill are not overpaid to a 
+level that results in a violation of rules 1 or 3 when the remaining portion is
+matched.
 
-====Penalties====
+===Penalties===
 
 The primary penalty for breeches of conduct is a '''ban''', which includes loss 
-of trading privileges, forfeiture of registration fee, and immediate cancelling 
-of any unfilled orders
+of trading privileges, forfeiture of registration fee, and immediate revocation
+of any unfilled orders. 
 
-===Data API===
+Less drastic punitive measures such as a cool-down period may be considered 
+for minor, first-time or infrequent conduct violations.
+
+==Data API==
 
 Trade history will be made available to both websocket and HTTP clients. 
 
 ==Exchange Administration==
 
-===Infrastructure===
+Operation of the DEX will require substantial administrative labor. 
+High throughput networking and high performance hardware are expected 
+requirements.
+Historical trade data should be archived for some reasonable amount of time, 
+if not forever.
+Clients will expect near-perfect uptime and long-term operation. 
```

Comment:
(seems like running a server behind Tor is out of the question... just an observation.)